sympy/physics/vector/functions.py

```diff
@@ -249,112 +251,122 @@ def kinematic_equations(speeds, coords, rot_type, rot_order=''):
     >>> q1, q2, q3 = dynamicsymbols('q1 q2 q3')
     >>> vprint(kinematic_equations([u1,u2,u3], [q1,q2,q3], 'body', '313'),
     ...     order=None)
-    [-(u1*sin(q3) + u2*cos(q3))/sin(q2) + q1', -u1*cos(q3) + u2*sin(q3) + q2', (u1*sin(q3) + u2*cos(q3))*cos(q2)/sin(q2) - u3 + q3']
+    [-u1*sin(q3) - u2*cos(q3) + sin(q2)*q1', -u1*cos(q3) + u2*sin(q3) + q2', (u1*sin(q3) + u2*cos(q3))*cos(q2) + (-u3 + q3')*sin(q2)]
 
     """
 
     # Code below is checking and sanitizing input
     approved_orders = ('123', '231', '312', '132', '213', '321', '121', '131',
                        '212', '232', '313', '323', '1', '2', '3', '')
-    rot_order = str(rot_order).upper()  # Now we need to make sure XYZ = 123
-    rot_type = rot_type.upper()
-    rot_order = [i.replace('X', '1') for i in rot_order]
-    rot_order = [i.replace('Y', '2') for i in rot_order]
-    rot_order = [i.replace('Z', '3') for i in rot_order]
-    rot_order = ''.join(rot_order)
+    # make sure XYZ = 123 and rot_type is in lower case
+    rot_order = translate(str(rot_order), 'XYZxyz', '123123')
+    rot_type = rot_type.lower()
 
     if not isinstance(speeds, (list, tuple)):
         raise TypeError('Need to supply speeds in a list')
     if len(speeds) != 3:
         raise TypeError('Need to supply 3 body-fixed speeds')
     if not isinstance(coords, (list, tuple)):
         raise TypeError('Need to supply coordinates in a list')
-    if rot_type.lower() in ['body', 'space']:
+    if rot_type in ['body', 'space']:
         if rot_order not in approved_orders:
             raise ValueError('Not an acceptable rotation order')
         if len(coords) != 3:
             raise ValueError('Need 3 coordinates for body or space')
         # Actual hard-coded kinematic differential equations
+        qid = q1d, q2d, q3d = [
+            diff(i, dynamicsymbols._t) for i in coords]
         q1, q2, q3 = coords
-        q1d, q2d, q3d = [diff(i, dynamicsymbols._t) for i in coords]
         w1, w2, w3 = speeds
         s1, s2, s3 = [sin(q1), sin(q2), sin(q3)]
         c1, c2, c3 = [cos(q1), cos(q2), cos(q3)]
-        if rot_type.lower() == 'body':
+        if rot_type == 'body':
             if rot_order == '123':
-                return [q1d - (w1 * c3 - w2 * s3) / c2, q2d - w1 * s3 - w2 *
```

Comment:
These equations were generated by sympy: I had it output eq.as_numer_denom()[0] for each equation, so if one is right they should all be right.